### PR TITLE
Feature/local storage

### PR DESCRIPTION
**Pull request by NikCrmr**

### CODE REVIEW

File: components/TextBox/TextBox.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { TextBox } from "./TextBox.styled";
+
+export default function StyledTextBox() {
+  return <TextBox name="text" placeholder="What's on your mind today?" />;
+}
```
Comment (MerleMak) [inline components/TextBox/TextBox.js:5]: just import Textbox directly to the form.js

File: pages/_app.js [hunk 2]
```diff
@@ -1,10 +1,50 @@
 import GlobalStyle from "../styles";
+import { useState } from "react";
+import { nanoid } from "nanoid";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function App({ Component, pageProps }) {
+  const [rangeValue, setRangeValue] = useState(127.5);
+  // const [allEntries, setAllEntries] = useState([]);
```
Comment (MerleMak) [inline pages/_app.js:8]:
```suggestion
```

File: utils/onSubmit.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    text: event.target.text.value,
+    slider: rangeValue,
+
+    date: new Date().toLocaleString(),
+  };
+
+  setAllEntries([...allEntries, submittedForm]);
+  console.log([...allEntries, submittedForm]);
+  event.target.reset();
+}
```
Comment (MerleMak) [inline utils/onSubmit.js:21]: You can delete this. 
You have already implemented this code in the _app.js, so I don't think you are using it anywhere

File: pages/flow.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    ))}
+                </StyledExperienceText>
+                <span>. You wrote:</span>{" "}
+                <StyledEntryText>{entry.text}</StyledEntryText>
+              </p>
+              <StyledSeparator />
+            </StyledEntry>
+          );
+        })}
+      </StyledEntryList> */}
```
Comment (MerleMak) [inline pages/flow.js:44]:
```suggestion
```
always delete commented out code before the commit!

File: pages/flow.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+// const StyledEntryText = styled.span`
+//   font-weight: bold;
+//   font-style: italic;
+// `;
+
+// const StyledSeparator = styled.hr`
+//   border: 0.5px solid grey;
+//   width: 100%;
+// `;
```
Comment (MerleMak) [inline pages/flow.js:75]:
```suggestion
```
!

File: pages/_app.js [hunk 6]
```diff
@@ -1,10 +1,50 @@
...
+        { sadness: event.target.sadness.checked },
+      ],
+      text: event.target.text.value,
+      slider: rangeValue,
+
+      date: new Date().toLocaleString(),
+    };
+
+    setAllEntries([...allEntries, submittedForm]);
+    console.log([...allEntries, submittedForm]);
```
Comment (MerleMak) [inline pages/_app.js:35]:
```suggestion
```

File: pages/_app.js [hunk 7]
```diff
@@ -1,10 +1,50 @@
...
+    defaultValue: [],
+  });
+
+  const handleRangeChange = (event) => {
+    setRangeValue(event.target.value);
+  };
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const submittedForm = {
```
Comment (MerleMak) [inline pages/_app.js:19]: beautiful!!

File: data.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+[
+  {
+    "id": 1,
+    "slider-value": 200
+  }
+]
```
Comment (MerleMak) [inline data.json:6]: i think you can delete this?

**Review by MerleMak — CHANGES_REQUESTED** [review]
Very long review 😮‍💨
But here you go! I left you some structural feedback and some comments and console.log that you forgot :P 
Remember to always delete them before committing your code. 

One thing that i quite important:
This PR has almost nothing to do with Local Storage. The local storage her eis like one line of code.

You did EVERYTHING here, so in the future try to seperate all these tasks into different tickets and you youll be good to go! :)


Nice job on the code otherwise! I know it seems like a lot of comments, but it is absolutely normal with this many lines of code. 

Keep it going guys! Your are on a good way 👌🏻

File: components/EntryList/EntryList.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+import { EntryList } from "./EntryList.styled";
+
+export default function StyledEntryList({ allEntries }) {
+  console.log("styled entrylist", allEntries);
+  return (
+    <EntryList>
+      <Entry allEntries={allEntries} />
+    </EntryList>
+  );
+}
```
[the remark this answers is not in the document]
Comment (Ramin-faqyri) [reply components/EntryList/EntryList.js:11]: We don't have a entry component yet to display the details. We resolved the issue like you suggested in the previous comment.

**Review by MerleMak — APPROVED** [review]

**Pull request merged**